pkg/controller/registry/reconciler/grpc.go

```diff
@@ -523,6 +545,29 @@ func imageChanged(logger *logrus.Entry, updatePod *corev1.Pod, servingPods []*co
 	return false
 }
 
+func isPodDead(pod *corev1.Pod) bool {
+	for _, check := range []func(*corev1.Pod) bool{
```

Comment:
This is a clever way to provide an extensible approach to a series of checks... but do we need it?  It seems we would have the same benefits by doing `s/isPodDead/IsPodDeletedByTaintManager/g`

Reply:
I'm somewhat confident we'll find another way for pods to be dead. We've seen similar issues in `operator-lib` leader for life.

So I figured I'd make things super easy for ourselves next time around.

Reply:
Yeah, do we think we'll need to expand this in the future?

Reply:
Well, I ain't agin' it, but I generally like to supply it when it's needed. 
/lgtm
from me

Reply:
I'm not objecting... just asking.